scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,18 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                kwargs['flags'] = response.flags
+                if self.keep_encoding_header:
+                    if isinstance(kwargs.get('flags'), list):
```

Comment:
Now that you set `kwargs['flags'] = response.flags` beforehand, this type check is not necessary, it will always be a list. See https://github.com/scrapy/scrapy/blob/master/scrapy/http/response/__init__.py#L25

Reply:
Dang it. Thank you for helping guide me through this by the way too!